src/Skia/Avalonia.Skia/Helpers/ImageSavingHelper.cs

```diff
@@ -40,25 +48,67 @@ public static void SaveImage(SKImage image, string fileName, int? quality = null
         /// The quality value is interpreted from 0 - 100. If quality is null 
         /// the encoder applies the default quality value.
         /// </param>
+        [Obsolete("Use overloads that take format parameter")]
         public static void SaveImage(SKImage image, Stream stream, int? quality = null)
         {
-            if (image == null) throw new ArgumentNullException(nameof(image));
-            if (stream == null) throw new ArgumentNullException(nameof(stream));
+            if (image == null)
+                throw new ArgumentNullException(nameof(image));
+            if (stream == null)
+                throw new ArgumentNullException(nameof(stream));
 
-            if (quality == null)
+            using var data = image.Encode();
+            data.SaveTo(stream);
+        }
+
+        /// <summary>
+        /// Save Skia image to a stream.
+        /// </summary>
+        /// <param name="image">Image to save</param>
+        /// <param name="stream">The output stream to save the image.</param>
+        /// <param name="format">The file format used to encode the image.</param>
+        /// <param name="quality">
+        /// The optional quality setting for image encoder. 
+        /// The quality value is interpreted from 0 - 100. If quality is null 
+        /// the encoder applies the default quality value. 
+        /// Some formats completely ignore this value.
+        /// </param>
+        public static void SaveImage(SKImage image, Stream stream, SKEncodedImageFormat format, int? quality = null)
+        {
+            if (image == null)
+                throw new ArgumentNullException(nameof(image));
+            if (stream == null)
+                throw new ArgumentNullException(nameof(stream));
+            if (quality < 0 || quality > 100)
+                throw new ArgumentOutOfRangeException(nameof(quality), "[0, 100]");
+
+            using var data = image.Encode(format, quality ?? 100);
+            data.SaveTo(stream);
+        }
+
+        private static SKEncodedImageFormat GetFormatFromExtension(string extension)
+            => extension.ToLowerInvariant() switch
             {
-                using (var data = image.Encode())
-                {
-                    data.SaveTo(stream);
-                }
-            }
-            else
+                ".png" => SKEncodedImageFormat.Png,
```

Comment:
Have you _actually_ checked if those work? SkiaSharp doesn't come compiled with those encoders

Reply:
Commented out the ones that do not have encoders by default.